### PR TITLE
chore: removing warning alternative color from codebase

### LINKED ISSUE #24966: chore: remove all instances of deprecated warning colors
### What is this about?

In preparation for upgrading the `@metamask/design-tokens` library to version 4, we need to remove all instances of the deprecated `--color-warning-alternative` and `--color-warning-disabled`. This color has been removed in v4 and its presence will cause breakages in the application.

[Design tokens migration v3 => v4 documentation](https://github.com/MetaMask/design-tokens/blob/main/MIGRATION.md#from-version-300-to-400)

### Scenario

_No response_

### Design

_No response_

### Technical Details

- Search the codebase for all instances of `--color-warning-alternative` and `--color-warning-disabled.
- Replace `--color-warning-alternative` and `--color-warning-disabled with an appropriate alternative from the updated design tokens.
- Ensure all components using this color are updated accordingly and tested for visual consistency.
- Update any documentation or style guides referencing `--color-warning-alternative` and `--color-warning-disabled.

### Threat Modeling Framework

_No response_

### Acceptance Criteria

- All instances of `--color-warning-alternative` and `--color-warning-disabled are removed from the codebase.
- The application functions correctly without any references to `--color-warning-alternative` and `--color-warning-disabled.
- Visual appearance of components previously using `--color-warning-alternative` and `--color-warning-disabled is consistent with design specifications.
- No breaking changes introduced in the application due to this update.

### Stakeholder review needed before the work gets merged

- [X] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### PR DESCRIPTION
**Pull request by georgewrmarshall**
## **Description**

This pull request removes the deprecated warning colors(`warning-alternative` and `warning-disabled`) in preparation for upgrading to [design token v4](https://github.com/MetaMask/design-tokens/blob/main/MIGRATION.md#from-version-300-to-400).

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25010?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/24966

## **Manual testing steps**

1. Go to the storybook build of this PR
2. Go to all stories that did use the warning alternative:
3. See color updates are working as expected
4. Ensure not instances of `warning-alternative`, `warning-disabled` exists

## **Screenshots/Recordings**

All before/after screenshots and screencasts have been moved to comments for easier review. This screenshot shows no instances of warning-alternative CSS variables or enums left in the codebase

Screenshot 2024-06-03 at 2 26 34 PM
Screenshot 2024-06-03 at 2 26 17 PM

Screenshot 2024-06-03 at 2 32 41 PM
Screenshot 2024-06-03 at 2 32 53 PM

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/app/confirm/info/row/value-double.tsx [hunk 1]
```diff
@@ -19,7 +19,7 @@ export type ConfirmInfoRowValueDoubleProps = {
 const LEFT_TEXT_COLORS = {
   [ConfirmInfoRowVariant.Default]: TextColor.textMuted,
   [ConfirmInfoRowVariant.Critical]: Color.errorAlternative,
-  [ConfirmInfoRowVariant.Warning]: Color.warningAlternative,
+  [ConfirmInfoRowVariant.Warning]: Color.warningDefault,
 };
```
Comment (georgewrmarshall) [inline ui/components/app/confirm/info/row/value-double.tsx:23]: ### After

Screenshot 2024-06-03 at 2 37 39 PM

File: ui/components/app/confirm/info/row/row.tsx [hunk 2]
```diff
@@ -59,7 +59,7 @@ const TOOLTIP_ICONS = {
 const TOOLTIP_ICON_COLORS = {
   [ConfirmInfoRowVariant.Default]: Color.iconMuted,
   [ConfirmInfoRowVariant.Critical]: Color.errorAlternative,
-  [ConfirmInfoRowVariant.Warning]: Color.warningAlternative,
+  [ConfirmInfoRowVariant.Warning]: Color.warningDefault,
 };
```
Comment (georgewrmarshall) [inline ui/components/app/confirm/info/row/row.tsx:63]: ### After

Screenshot 2024-06-03 at 2 36 37 PM

File: .storybook/3.COLORS.stories.mdx [hunk 3]
```diff
@@ -99,10 +99,8 @@ var(--color-error-disabled) [DEPRECATED]
 
 /** Warning */
 var(--color-warning-default)
-var(--color-warning-alternative) [DEPRECATED]
 var(--color-warning-muted)
 var(--color-warning-inverse)
-var(--color-warning-disabled) [DEPRECATED]
```
Comment (georgewrmarshall) [inline .storybook/3.COLORS.stories.mdx:105]: ### After


https://github.com/MetaMask/metamask-extension/assets/8112138/c8f95fe5-9320-491b-b87e-ffc255b2802f

File: ui/components/component-library/header-base/header-base.stories.tsx [hunk 4]
```diff
@@ -298,7 +298,7 @@ export const UseCaseDemos = (args) => (
       </HeaderBase>
     </Box>
     <Text>startAccessory and endAccessory assigned </Text>
-    <Box backgroundColor={BackgroundColor.warningAlternative}>
+    <Box backgroundColor={BackgroundColor.warningDefault}>
       <HeaderBase
         marginBottom={4}
         startAccessory={
```
Comment (georgewrmarshall) [inline ui/components/component-library/header-base/header-base.stories.tsx:304]: ### After

Screenshot 2024-06-03 at 2 38 49 PM

File: ui/components/multichain/pages/page/page.stories.tsx [hunk 5]
```diff
@@ -142,7 +142,7 @@ export const ScrollingStory = (args) => (
       <Content background={BackgroundColor.successAlternative}>
         {scrollingContent}
       </Content>
-      <Footer backgroundColor={BackgroundColor.warningAlternative}>
+      <Footer backgroundColor={BackgroundColor.warningDefault}>
         <ButtonPrimary block disabled>
           Cancel
         </ButtonPrimary>
```
Comment (georgewrmarshall) [inline ui/components/multichain/pages/page/page.stories.tsx:148]: ### After

Screenshot 2024-06-03 at 2 39 58 PM
Screenshot 2024-06-03 at 2 40 15 PM

File: ui/css/design-system/_colors.scss [hunk 6]
```diff
@@ -23,10 +23,8 @@ $color-map: (
   'error-inverse': --color-error-inverse,
   'error-disabled': --color-error-disabled,
   'warning-default': --color-warning-default,
-  'warning-alternative': --color-warning-alternative,
```
Comment (georgewrmarshall) [inline ui/css/design-system/_colors.scss:26]: Screenshot 2024-06-03 at 2 26 34 PM

File: ui/css/design-system/_colors.scss [hunk 7]
```diff
@@ -23,10 +23,8 @@ $color-map: (
...
   'warning-muted': --color-warning-muted,
   'warning-inverse': --color-warning-inverse,
-  'warning-disabled': --color-warning-disabled,
```
Comment (georgewrmarshall) [inline ui/css/design-system/_colors.scss:29]: Screenshot 2024-06-03 at 2 32 41 PM

File: ui/helpers/constants/design-system.ts [hunk 8]
```diff
@@ -29,10 +29,8 @@ export enum Color {
   errorInverse = 'error-inverse',
   errorDisabled = 'error-disabled',
   warningDefault = 'warning-default',
-  warningAlternative = 'warning-alternative',
```
Comment (georgewrmarshall) [inline ui/helpers/constants/design-system.ts:32]: Screenshot 2024-06-03 at 2 26 17 PM

File: ui/helpers/constants/design-system.ts [hunk 9]
```diff
@@ -29,10 +29,8 @@ export enum Color {
...
   warningMuted = 'warning-muted',
   warningInverse = 'warning-inverse',
-  warningDisabled = 'warning-disabled',
```
Comment (georgewrmarshall) [inline ui/helpers/constants/design-system.ts:35]: Screenshot 2024-06-03 at 2 32 53 PM

File: ui/components/ui/button/buttons.scss [hunk 10]
```diff
@@ -282,12 +282,12 @@ input[type="submit"][disabled] {
...
     &[disabled] {
-      border-color: var(--color-warning-alternative);
+      border-color: var(--color-warning-default);
       color: var(--color-text-muted);
     }
 
     &:active {
-      border-color: var(--color-warning-alternative);
+      border-color: var(--color-warning-default);
     }
```
Comment (georgewrmarshall) [inline ui/components/ui/button/buttons.scss:291]: ### After


https://github.com/MetaMask/metamask-extension/assets/8112138/94b5d6af-ce90-4179-97a0-20a76a2cb255

File: ui/helpers/constants/design-system.ts [hunk 11]
```diff
@@ -71,7 +69,6 @@ export enum BackgroundColor {
   errorAlternative = 'error-alternative',
   errorMuted = 'error-muted',
   warningDefault = 'warning-default',
-  warningAlternative = 'warning-alternative',
```
Comment (georgewrmarshall) [inline ui/helpers/constants/design-system.ts:74]: Same as above

File: ui/helpers/constants/design-system.ts [hunk 12]
```diff
@@ -99,7 +96,6 @@ export enum BorderColor {
...
```
Comment (georgewrmarshall) [inline ui/helpers/constants/design-system.ts:102]: Same as above

File: ui/components/app/confirm/info/row/row.tsx (hunk 2 above)
Comment (brianacnguyen) [reply ui/components/app/confirm/info/row/row.tsx:63]: can you show an example with an icon?

File: ui/components/ui/button/buttons.scss (hunk 10 above)
Comment (brianacnguyen) [reply ui/components/ui/button/buttons.scss:291]: shouldn't the text be colored as well?

**Review by brianacnguyen — COMMENTED** [review]
a few small comments

File: ui/components/app/confirm/info/row/row.tsx (hunk 2 above)
Comment (georgewrmarshall) [reply ui/components/app/confirm/info/row/row.tsx:63]: Screenshot 2024-06-05 at 3 09 04 PM
Screenshot 2024-06-05 at 3 09 12 PM

File: ui/components/ui/button/buttons.scss (hunk 10 above)
Comment (georgewrmarshall) [reply ui/components/ui/button/buttons.scss:291]: Good question. This is from a deprecated button that I believe isn't used anywhere, so I'm just keeping it consistent with what is currently there. See Before screencast

Screenshot shows no usage of type warning for a button component 

Screenshot 2024-06-05 at 3 10 48 PM

File: ui/components/app/confirm/info/row/row.tsx (hunk 2 above)
Comment (brianacnguyen) [reply ui/components/app/confirm/info/row/row.tsx:63]: thanks!

File: ui/components/component-library/header-base/header-base.stories.tsx (hunk 4 above)
Comment (brianacnguyen) [reply ui/components/component-library/header-base/header-base.stories.tsx:304]: unblocking: i feel like these should use brand colors instead

**Review by brianacnguyen — APPROVED** [review]

**Review by darkwing — APPROVED** [review]

**Pull request merged**